duckduckgo/iOS
# improve onboarding experiment 1

Task/Issue URL: https://app.asana.com/0/392891325557410/1110624620486758
Tech Design URL: n/a
CC:

**Description**:

Shows new onboarding depending on the variant.  See tests for details.

Tests should be performed on a range of devices:

* iPhone SE
* iPhone 6 / 8
* iPhone X
* iPad

Note: ATB tests now pass an environment variable to override the variant for consistency, setting it to `sc`, so that UI tests don't need to be updated for every experiment.

**Steps to test this PR**:

Control group `sc`:
1. Set variant to the control group (or blank)
1. Onboarding should behave as it has done thus far

Summary `mq`:
1. Set the variant to the summary onboarding
1. The onboarding is now a single summary page with a "Starting Browsing" button
1. Tapping the button cross-dissolves to the home screen
1. Rotation not supported for phone devices, but is for iPad devices

Summary + Contextual `mr`:
1. As summary plus...
1. Set the variant to the contextual onboarding
1. The app loads straight in to the home view
1. The first time on the home view a contextual tip showing "search privately" message should show
1. The second time on the home view a contextual tip showing "customize" message should show
1. The first time browsing a none-DDG page the "privacy grade" tip should show 
1. The second time browsing a non-DDG page (could even just be tapping a link on the above page) the "fire button" tip should show
1. Tapping anywhere on the screen should dismiss the tip[1] while simultaneously allowing whatever interaction the user intended to do

[1] A delay of 1 second has been deliberately added to prevent accidental dismissal. 

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
